### PR TITLE
Support --ip argument when run the container

### PR DESCRIPTION
**Pull request by Zheaoli**
Fix: #882 
Signed-off-by: Zheao.Li <me@manjusaka.me>

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
Can we have an integration test?

**Comment by Zheaoli** [comment]
> Can we have an integration test?
> 
> 

Yes, I'm working on an integration test, so I have marked this PR as draft

**Comment by AkihiroSuda** [comment]
Please squash commits

File: cmd/nerdctl/run.go [hunk 1]
```diff
@@ -121,6 +121,8 @@ func setCreateFlags(cmd *cobra.Command) {
 	cmd.Flags().StringSlice("dns", nil, "Set custom DNS servers")
 	// publish is defined as StringSlice, not StringArray, to allow specifying "--publish=80:80,443:443" (compatible with Podman)
 	cmd.Flags().StringSliceP("publish", "p", nil, "Publish a container's port(s) to the host")
+	// FIXME: not support IPV6 yet
+	cmd.Flags().StringP("ip", "", "", "IPv4 address to assign to the container")
```
Comment (AkihiroSuda) [inline cmd/nerdctl/run.go:125]: Please mention this in the Command Reference in README.md

File: pkg/ocihook/ocihook.go [hunk 2]
```diff
@@ -321,6 +326,22 @@ func getPortMapOpts(opts *handlerOpts) ([]gocni.NamespaceOpts, error) {
...
 
+func getIPAddressOpts(opts *handlerOpts) ([]gocni.NamespaceOpts, error) {
+	if opts.containerIP != "" {
+		if rootlessutil.IsRootlessChild() {
+			return nil, fmt.Errorf("containerIP assignment is not supported in rootless mode")
+		}
+
+		return []gocni.NamespaceOpts{
+			gocni.WithLabels(map[string]string{
+				"IgnoreUnknown": "1", // Special tick for go-cni
```
Comment (AkihiroSuda) [inline pkg/ocihook/ocihook.go:337]: Please explain what is “special tick” in the comment lines

File: pkg/ocihook/ocihook.go [hunk 3]
```diff
@@ -321,6 +326,22 @@ func getPortMapOpts(opts *handlerOpts) ([]gocni.NamespaceOpts, error) {
 	return nil, nil
 }
 
+func getIPAddressOpts(opts *handlerOpts) ([]gocni.NamespaceOpts, error) {
+	if opts.containerIP != "" {
+		if rootlessutil.IsRootlessChild() {
+			return nil, fmt.Errorf("containerIP assignment is not supported in rootless mode")
```
Comment (AkihiroSuda) [inline pkg/ocihook/ocihook.go:332]: Can be another PR, but you can implement this for rootless mode too.
Not reachable from the host though

Comment (Zheaoli) [reply pkg/ocihook/ocihook.go:332]: I think I can make rootless support in single PR

File: cmd/nerdctl/run.go (hunk 1 above)
Comment (fahedouch) [inline cmd/nerdctl/run.go:125]: plz use `cmd.Flags().String`

File: cmd/nerdctl/run_network_linux_test.go [hunk 4]
```diff
@@ -395,3 +395,64 @@ func TestRunPort(t *testing.T) {
...
+	}{
+		{
+			ip:            "172.0.0.2",
+			shouldSuccess: true,
+			useNetwork:    true,
+		},
+		{
+			ip:            "192.0.0.2",
+			shouldSuccess: false,
+			useNetwork:    true,
```
[the remark this answers is not in the document]
Comment (Zheaoli) [reply cmd/nerdctl/run_network_linux_test.go:422]: I think we need

**Review by AkihiroSuda — APPROVED** [review]
Thanks
